### DISCUSSION TITLE
Plotting using gratia::draw

### DISCUSSION THREAD

**Discussion by rvaradhan** [gavinsimpson/gratia#294] [Q&A] [answered]
I fit a GAM model:  gam(y ~ trt, s(time, by=trt, method="REML"), where the response y is measured at times `time`.  The variable `trt` is a factor with 4 levels.  Using "gratia" I would like to plot the fitted GAM model along with the original data so that I can visually inspect the adequacy of the fit.  There are 4 panels in my plot corresponding to each level of "trt". How can I add the observed data points to each panel, in addition to the fitted smoother?

**Comment by rvaradhan** [comment]
I would like to plot on the original scale of the response rather than on the partial effect scale.

**Comment by gavinsimpson** [comment, marked as answer]
What you're looking for is to predict from the model. This is usually done via `predict()`, but `gratia::fitted_values()` can will facilitate this:

```
library("mgcv")
library("gratia")
library("ggplot2")

eg_data <- data_sim("eg4", n = 400, dist = "normal", scale = 2, seed = 1)
m <- gam(y ~ fac + s(x2, by = fac), data = eg_data, method="REML", family = gaussian())

ds <- data_slice(m, x2 = evenly(x2), fac = evenly(fac))
fv <- fitted_values(m)

fv |>
  ggplot(aes(x = x2, y = .fitted, group = fac)) +
  geom_ribbon(aes(ymin = .lower_ci, ymax = .upper_ci, fill = fac), alpha = 0.2) +
  geom_line(aes(colour = fac)) +
  geom_point(data = eg_data, aes(colour = fac, y = y)) +
  facet_wrap(~ fac)
```

![gratia-294](https://github.com/gavinsimpson/gratia/assets/514696/26c3d94b-9c71-4b53-98c3-3e41a1ae1936)

**Reply by rvaradhan** [reply]
Dear Gavin - this is terrific.  I was able to get beautiful plots!  Thank you very much.

**Reply by rvaradhan** [reply]
Gavin - Is it possible to extend your solution to GAMM object?  I tried this but it doesn't work.  Thanks for any suggestion.

**Reply by rvaradhan** [reply]
Sorry - it works fine for the GAMM object as well.

**Comment by kobby184** [comment]
I am plotting a multinomial GAM , one of my smooth terms is s(longitude, latitude). values seem to be on top of each other. Is there a way around it. Thank you

**Reply by kobby184** [reply]
I use draw(model, rug=FALSE)

**Reply by gavinsimpson** [reply]
It's probably just overplotting individual data points that are observed at the same location. I could reduce these to distinct values of the covariate when plotting. But usually, people either want this on or off.

**Reply by gavinsimpson** [reply]
In the development version, `draw(model)` will now only show the unique data values by default. See #327
